index.js

```diff
@@ -331,3 +602,55 @@ function destroy() {;
   return Promise.all(promises);
 }
 
+async function lookupConversations(client, calls) {
+  let i = 0;
+  const result = [];
+  const convIds = calls.map(call => call.convId);
+  if(!convIds.length){
+    return result;
+  }
+  const conversations = await client.getConversationsByIds(convIds);
+  conversations.forEach(conversation => {
```

Comment:
I really don't think the index with each call will always match up with each conversation returned by the Api. Also, you could use .map here as well to create a new result array